### PR TITLE
[explorer] batch queryEvent calls using the max page limit size of 1000 to fix APY calculations

### PR DESCRIPTION
**Pull request by williamrobertson13**
## Description 
When we fetch validator events in the Explorer, we say that the limit is `30 * numValidators` which on testnet is 2850. The problem is that the `queryEvents` API has a maximum page size of `1000`, so the limit we were providing was causing the API to error out.

To fix this, we can batch the API calls together when the limit exceeds the maximum page size. So if we did have 2850 events, we'd fetch 1000 -> fetch another 1000 -> fetch the remaining 850. One thing to note is that there are only 190 validator events (2 completed epochs * 95 validators) thus far, so this batching functionality isn't actually being used at the moment.

## Test Plan 
- Manual testing on devnet with:
   -  4 validators and a query limit of 1
   - 4 validators and a query limit of 1000
   - 4 validators with an event limit of 5 and a query limit of 1
- Manual testing on testnet with:
   - 95 validators and a query limit of 50
   - 95 validators and a query limit of 1000
   - 95 validators with an event limit of 5 and a query limit of 2 (limit truncation case)

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [X] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes
N/A

### CODE REVIEW

File: apps/core/src/hooks/useGetValidatorsEvents.ts [hunk 1]
```diff
@@ -11,24 +15,57 @@ type GetValidatorsEvent = {
...
+                // Do some validation at the runtime level for some extra type-safety
+                // https://tkdodo.eu/blog/react-query-and-type-script#type-safety-with-the-enabled-option
+                throw new Error(
+                    `Limit needs to always be defined! Received ${limit} instead.`
+                );
+            }
+
+            if (limit > QUERY_MAX_RESULT_LIMIT) {
+                let hasNextPage = true;
+                let currCursor = cursor;
```
Comment (Jordan-Mysten) [inline apps/core/src/hooks/useGetValidatorsEvents.ts:43]: The outer cursor param never seems used, can we remove it to simplify?

File: apps/core/src/hooks/useGetValidatorsEvents.ts [hunk 2]
```diff
@@ -11,24 +15,57 @@ type GetValidatorsEvent = {
...
+                const results: SuiEvent[] = [];
+
+                while (hasNextPage) {
```
Comment (Jordan-Mysten) [inline apps/core/src/hooks/useGetValidatorsEvents.ts:46]: You also need to stop paging if you hit the limit, this will page out the entire list which isn't desired.

File: apps/core/src/hooks/useGetValidatorsEvents.ts (hunk 1 above)
Comment (williamrobertson13) [reply apps/core/src/hooks/useGetValidatorsEvents.ts:43]: yep

File: apps/core/src/hooks/useGetValidatorsEvents.ts [hunk 3]
```diff
@@ -2,33 +2,56 @@
...
+                        limit: Math.min(limit!, QUERY_MAX_RESULT_LIMIT),
+                        order,
+                    });
+
+                    hasNextPage = validatorEventsResponse.hasNextPage;
+                    currCursor = validatorEventsResponse.nextCursor;
+                    results.push(...validatorEventsResponse.data);
+                }
+                return results;
+            }
```
Comment (Jordan-Mysten) [inline apps/core/src/hooks/useGetValidatorsEvents.ts:46]: One final note here is that this API can now return more than the limit. I'd truncate the limit at the end just to avoid that.

**Review by Jordan-Mysten — APPROVED** [review]

File: apps/core/src/hooks/useGetValidatorsEvents.ts [hunk 4]
```diff
@@ -11,24 +15,57 @@ type GetValidatorsEvent = {
...
-    // since we are getting events base on the number of validators, we need to make sure that the limit is not null and cache by the limit
-    // number of validators can change from network to network
+    // Since we are getting events based on the number of validators, we need to make sure that the limit
+    // is not null and cache by the limit number of validators can change from network to network
     return useQuery(
         ['validatorEvents', limit, cursor, order],
-        () =>
-            rpc.queryEvents({
+        async () => {
+            if (limit === null) {
```
[the remark this answers is not in the document]
Comment (williamrobertson13) [reply apps/core/src/hooks/useGetValidatorsEvents.ts]: kk I totally misread this comment, `!limit` vs `limit === null` and not `limit!` 😶‍🌫️

File: crates/sui-framework/src/natives/crypto/ecdsa_k1.rs [hunk 5]
```diff
@@ -106,15 +106,14 @@ pub fn ecrecover(
 
     let cost = context.gas_used();
 
-    let sig = match <Secp256k1RecoverableSignature as ToFromBytes>::from_bytes(&signature_ref) {
```
Comment (oxade) [inline crates/sui-framework/src/natives/crypto/ecdsa_k1.rs:109]: Seems like a lot of unrelated changes made their way in?
